internal/snapshot/download.go

```diff
@@ -0,0 +1,72 @@
+/*
+ * Flow CLI
+ *
+ * Copyright 2019-2021 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package snapshot
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+
+	"github.com/onflow/flow-cli/internal/command"
+	"github.com/onflow/flow-cli/pkg/flowkit"
+	"github.com/onflow/flow-cli/pkg/flowkit/services"
+)
+
+type flagsDownload struct {
+	OutputFileName string `default:"root-protocol-state-snapshot.json" flag:"output-filename" info:"The name for the JSON output file"`
+	OutputDir      string `default:"./" flag:"output-dir" info:"The directory to output the protocol state snapshot"`
```

Comment:
I second this, and I would even say that you should reuse the common output flag we use in the CLI for saving `--save` at the same time this begs the question of flag vs argument, I've written CLI guidelines where I define when to use which https://github.com/onflow/flow-cli/blob/master/CONTRIBUTING.md#arguments 
In your case if the command can print the details to screen as well then use optional `--save` flag, if command must always save to a file then use argument, because flags by their nature indicate they are optional whereas arguments are required.

Reply:
Will change download command to get  @sideninja and combine the 2 flags @jordanschalm

Reply:
Also it is not useful to display the contents of the protocol snapshot to the screen because the output is large and hard to reason about in the terminal. Strictly outputting to specified file is the cleanest UX we can provide.

Reply:
That is completely understandable then, but I would advise switching the command to require this path as an argument, `flow snapshot save {location}`  or you can even leave download, if we are directly saving and that is the only behaviour then that's good. I wouldn't use a flag for save location because the command requires it.

Reply:
The issue with using the builtin save flag is that before outputResult is invoked the result is [formatted](https://github.com/onflow/flow-cli/blob/fa2ee709a71fed90f17cb37799872e344484cb98/internal/command/command.go#L124) . We avoid formatting the protocol snapshot because we cannot unmarshal the byte representation which is why I am only returning the output location as the DownloadResult .

Reply:
Is there a way to disable the json output format for a specific command? We would need to write the bytes directly to the output file , so that we avoid failing [here](https://github.com/onflow/flow-cli/blob/fa2ee709a71fed90f17cb37799872e344484cb98/internal/command/result.go#L73)

Reply:
If we add relic to the build process :eyes: then we can parse the snapshot bytes in JSON use the standard save flag

Reply:
No no I'm saying that in case if this doesn't make sense to be displayed to the screen (which you said it doesn't) then you should stay with your idea of the command saving to the file, but I'm just saying that a command argument feels more appropriate than a flag to define the location where to save to. Also get doesn't make that much sense if the command always saves, it can stay "download" or "save". So the command should look like: `flow snapshot save ./root-snapshot.json` (note passing the location as argument instead of flag).

Reply:
Ahhh Okay I see , I misunderstood your previous comment .

Reply:
https://github.com/onflow/flow-cli/pull/440/commits/cff23dc6ca08f838b03aa54456e61bfb2a3d9b54 @sideninja